src/Documenter.jl

```diff
@@ -107,6 +108,20 @@ written when [`makedocs`](@ref) is run. The name of the build directory is, by c
 called `build`, though, like with `source`, users are free to change this to anything else
 to better suit their project needs.
 
+**`workdir`** determines the working directory where `@example` and `@repl` code blocks are
+executed. It can be either a path or the special value `:build` (default).
+
+If the `workdir` is set to a path, the working directory is reset to that path for each code
+block being evaluated. Relative paths are taken to be relative to `root`, but using absolute
+paths is recommended (e.g. `workdir = joinpath(@__DIR__, "..")` for executing in the package
+root for the usual `docs/make.jl` setup).
+
+With the default `:build` option, the working directory is set to a subdirectory of `build`,
+determined from the source file path. E.g. for `src/foo.md` it is set to `build/`, for
+`src/foo/bar.md` it is set to `build/foo` etc.
```

Comment:
I took the liberty of editing the docs a little bit. Do you think this is clear enough do you think it would be better to have the slightly more explicit example you had before?

Reply:
I think this is clearer than my example actually, thanks!